src/daft-plan/src/planner.rs

```diff
@@ -80,17 +80,24 @@ pub fn plan(logical_plan: &LogicalPlan) -> DaftResult<PhysicalPlan> {
                 scan_op,
                 ..
             })) => {
-                let scan_tasks = scan_op
-                    .0
-                    .to_scan_tasks(pushdowns.clone())?
-                    .collect::<DaftResult<Vec<_>>>()?;
+                // Currently naively concats all ScanTasks together into one fat ScanTask
+                // TODO: we can instead forward this iterator of ScanTasks to the physical planner which can
+                // perform intelligent coalescing/splitting of ScanTasks to be right-sized for the executor
+                let coalesced_scan_task = ScanTask::concat(
```

Comment:
I'm having a hard time understanding why we are combining everything into a single scan task. Doesn't that defeat the purpose of what we are trying to do with metadata and statistics?

Reply:
Fixed and now the Python API expects a `list[ScanTask]` instead